src/app/workspace/workspace.service.ts

```diff
@@ -114,4 +115,46 @@ export class WorkspaceService {
     const user = await this.userRepository.findOneBy({ email });
     return { userExists: user ? true : false };
   }
+  
+  public async createWorkspace(createWorkspaceDto: CreateWorkspaceDto, user: User ):
+  Promise<{projectName: string, settings: string, ownerId: number}> {
+    const workspace = { ...createWorkspaceDto, ownerId: user.id};
+    // const workspace = this.workspaceRepository.create(createWorkspaceDto);
+    return this.workspaceRepository.save(workspace);
+  }
+
+  async findAllWorkspaces(): Promise<Workspace[]> {
+    return await this.workspaceRepository.find();
+  }
+
+  async findOneWorkspace(id: number): Promise<Workspace> {
+    const workspace = await this.workspaceRepository.findOneBy({ id });
+    if ( workspace.deletedAt === null ) {
```

Comment:
This condition will be true if workspace is't deleted, you probably want to check if deletedAt is not null.